src/js/firstparties/facebook.js

```diff
@@ -0,0 +1,63 @@
+// Adapted from https://github.com/mgziminsky/FacebookTrackingRemoval
+(function() {
+let fb_wrapped_link = "a[href*='facebook.com/l.php?'";
+
+function findInAllFrames(query) {
+  let out = [];
+  document.querySelectorAll(query).forEach((node) => {
+    out.push(node);
+  });
+  Array.from(document.getElementsByTagName('iframe')).forEach((iframe) => {
+    try {
+      iframe.contentDocument.querySelectorAll(query).forEach((node) => {
+        out.push(node);
+      });
+    } catch (e) {
+      // pass on cross origin iframe errors
+    }
+  });
+  return out;
+}
+
+// remove all attributes from a link except for class and ARIA attributes
+function cleanAttrs(elem) {
+  for (let i = elem.attributes.length - 1; i >= 0; --i) {
+    const attr = elem.attributes[i];
+    if (attr.name !== 'class' && !attr.name.startsWith('aria-')) {
+      elem.removeAttribute(attr.name);
+    }
+  }
+}
+
+// Remove excessive attributes and event listeners from link a and replace
+// its destination with href
+function cleanLink(a) {
+  let href = new URL(a.href).searchParams.get('u');
```

Comment:
What if Facebook keeps the "l.php" href nonsense (so that our selector still matches) but changes something about destination URL replacement so that we end up breaking all links on Facebook? We should add guard clauses to confirm things are what they expect them to be before making any changes to the page. Simply checking that `href` is truthy should already be an improvement.